vars/randomString.groovy

```diff
@@ -0,0 +1,45 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+/**
+Generate a random string
+
+  // Create a random string of 15 chars (alphanumeric)
+  def value = randomString(size: 15)
+*/
+
+import java.security.SecureRandom
+import java.util.Base64
+
+def call(Map args = [:]) {
+  def size = args.get('size', 128)
```

Comment:
I don't think we should allow an unlimited size here. We should error out if it's bigger than something reasonable, like 128K.

Reply:
My initial plan was, to leave the default one and let the consumers to set the size:

For instance:

- There is a terraform validation for some variables -> https://github.com/elastic/azure-vm-extension/blob/c5a89c1480940dac7f4e3ca840716cd9518521e4/test/terraform/variables.tf#L19-L50

The format is opinionated and therefore cannot contain certain chars, otherwise:

```
[2021-06-10T10:46:31.098Z] │ Error: "name" must end with an alphanumeric character or underscore
[2021-06-10T10:46:31.098Z] │ 
[2021-06-10T10:46:31.098Z] │   with azurerm_windows_virtual_machine.main,
[2021-06-10T10:46:31.098Z] │   on main.tf line 36, in resource "azurerm_windows_virtual_machine" "main":
[2021-06-10T10:46:31.099Z] │   36: resource "azurerm_windows_virtual_machine" "main" {
```

Then the changes in the consumer would look like -> https://github.com/elastic/azure-vm-extension/pull/13/files#r649107174

We can probably make the step more generic if needed. The requirement was:
- alphanumeric with -
- never ending in - but some alphanumeric
- size variable